cli_bot.py

```diff
@@ -0,0 +1,433 @@
+    def birthdays(self, *args: str) -> str:
+        """Prepares all contacts to be outputted into console that have BD in a following week.
+
+        Args:
+            args: Command doesn't expect any args, list should be empty.
+
+        Returns:
+            Command output.
+        """
+        command_output = ""
+        if len(args) != 1:
+            raise CommandOperationalError (
+                "Warning: Command expect one argument: number of days. "
+                f"Received: {' '.join(args)}\n"
+            )
+        days = int(args[0])
+
+        results = self._address_book.get_birthdays_per_days(days).items()
+        if not results:
+            return "No contacts found"
+
+        command_output += "Contacts per day: "
+        for date, records in results:
+            records_str = '| ' + '\n | '.join(str(r) for r in records)
+            command_output += f"\nHave BD on {date}:\n {records_str}"
+
+        return command_output
+
+    def print_all_contacts(self, *args: str) -> str:
+        """Prepares contacts to be outputted into console.
+
+        Args:
+            args: Command doesn't expect any args, list should be empty.
+
+        Returns:
+            Command output.
+        """
+        command_output = ""
+        if args:
+            command_output += (
+                "Warning: Command doesn't expect any arguments. "
+                f"Received: {' '.join(args)}\n"
+            )
+
+        command_output += "All Records: "
+        for record in self._address_book.values():
+            command_output += f"\n{record}"
+
+        return command_output
+
+    @input_error(error_msg_base="Command execution failed")
+    def execute_command(self, command: str, args: list[str]) -> str:
+        if command not in self.supported_commands:
+            raise CommandNotSupported(f"command '{command}' is not supported!")
+
+        command_handler = self.supported_commands[command]
+        return command_handler(*args)
+
+    @input_error(error_msg_base="Command 'add-address' failed")
+    def add_address(self, *args: str):
+        """Add address to already existing record.
+
+        Args:
+            args: List with username and address.
+
+        Returns:
+            Command output
+
+        Raises:
+            CommandOperationalError if user does not exist.
+            ValueError if date is invalid.
+        """
+        if len(args) != 2:
+            raise CommandOperationalError(
+                "command expects an input of two arguments: username and address, separated by a space. "
+                f"Received: {' '.join(args)}"
+            )
+        username, address_str = args
+        try:
+            record = self._address_book.find(username)
+
+        except KeyError as e:
+            raise CommandOperationalError(
+                f"user with username {username} doesn't exist. "
+                f"If you want to add address, please use 'add-address' command."
```

Comment:
This message looks sus, he already calls this command, maybe this supposed to be reference to 'add' command to add a user that's being looked here ?